scenario/upgrade.py

```diff
@@ -0,0 +1,28 @@
+"""
+Scenario: base
+
+Generic scenario that tries many tests on a platform with or without a relay.
+This is the base scenario that should be run to test a release.
+"""
+from scenario.lib import *
+import os
+
+# test begins, register start time
+start(__doc__)
+
+import scenario.accept_nodes as accept_nodes
+
+import scenario.create_user as create_user
+
+# Test base rudder and accept nodes
+accept_nodes.execute()
+create_user.execute("test_user1", "3188d0c4-db97-4a8d-bb05-2e8061b991b8")
+# Upgrade
+test_shell_on("server", '/usr/local/bin/rudder-setup upgrade-server 6.0-nightly', error_mode=Err.CONTINUE, live_output=True)
+# Update API token and re-test the webapp
+(rudder_url, token) = scenario.platform.api_connection_info()
+os.environ['RUDDER_TOKEN'] = token
```

Comment:
This should not work, os.environ is read only, writing to it doesn't affect environment

Reply:
Ok it seems to be used as the default for sub process at least